Source/WebCore/css/StyleProperties.cpp

```diff
@@ -1118,15 +1202,33 @@ String StyleProperties::borderRadiusShorthandValue(const StylePropertyShorthand&
     return result.toString();
 }
 
-String StyleProperties::borderPropertyValue(const StylePropertyShorthand& width, const StylePropertyShorthand& style, const StylePropertyShorthand& color) const
+String ShorthandSerializer::serializeBorder(unsigned sectionLength) const
 {
-    auto widthText = getCommonValue(width);
+    auto sectionsTotalLength = sectionLength * 3;
+    ASSERT(sectionsTotalLength <= length());
+
+    if (!subsequentLonghandsHaveInitialValues(sectionsTotalLength))
+        return String();
+
+    // Serializable properties are widths, then styles, then colors.
+    auto serializeSection = [&](unsigned sectionStartIndex) {
+        String result;
+        for (unsigned i = 0; i < sectionLength; ++i) {
+            String text = serializeLonghandValue(sectionStartIndex + i);
+            if (result.isNull())
+                result = text;
+            else if (result != text)
+                return String();
+        }
+        return result;
+    };
+    auto widthText = serializeSection(0);
     if (widthText.isNull())
         return String();
-    auto styleText = getCommonValue(style);
+    auto styleText = serializeSection(sectionLength);
     if (styleText.isNull())
         return String();
-    auto colorText = getCommonValue(color);
+    auto colorText = serializeSection(sectionLength + sectionLength);
```

Comment:
Nit: you are using `* 3` above, so this could use `* 2`.

Reply:
OK